### PR TITLE
fix(purpur): apply log4j mitigations specific to Purpur

### LINKED ISSUE #1274: Log4j Patch - Wrong flags for version
### Describe the problem

I'm running a Minecraft 1.16.5 Purpur server and the Log4j JVM flags don't match the official Minecraft instructions  
> 1.12-1.16.5: Download [this file](https://launcher.mojang.com/v1/objects/02937d122c86ce73319ef9975b58896fc1b491d1/log4j2_112-116.xml) to the working directory where your server runs. Then add the following JVM arguments to your startup command line: 
-Dlog4j.configurationFile=log4j2_112-116.xml

However, my JVM flags are the following:
```
java -XX:+UseShenandoahGC -XX:+AlwaysPreTouch -XX:ShenandoahGCHeuristics=compact -XX:ConcGCThreads=4 -XX:+UseLargePages -XX:+DisableExplicitGC -XX:+UseLargePagesInMetaspace -Xmx12G -Xms12G -Dlog4j2.formatMsgNoLookups=true -jar purpur-1.16.5-1171.jar
```
This Log4j configuration flag was actually provided by Minecraft for 1.17 servers

### Container definition

```yml
version: "3"

services:
  mc:
    image: itzg/minecraft-server:java11
    ports:
      - 25565:25565
    environment:
      EULA: "TRUE"
      VERSION: "1.16.5"
      TYPE: "PURPUR"
      MEMORY: "12G"
      JVM_XX_OPTS: "-XX:+UseShenandoahGC -XX:+AlwaysPreTouch -XX:ShenandoahGCHeuristics=compact -XX:ConcGCThreads=4 -XX:+UseLargePages -XX:+DisableExplicitGC"
      TZ: "Europe/London"
      USE_LARGE_PAGES: "true"
      MODE: "creative"
      SPAWN_PROTECTION: "0"
      VIEW_DISTANCE: "8"
      ENABLE_RCON: "true"
      RCON_PASSWORD: "rcon_passwd_redacted"
    network_mode: "host"

    tty: true
    stdin_open: true
    restart: unless-stopped
    volumes:
      # attach a directory relative to the directory containing this compose file
      - /opt/minecraft/cobalt:/data
  backups:
    image: itzg/mc-backup
    environment:
      BACKUP_INTERVAL: "24h"
      RCON_PASSWORD: "rcon_passwd_redacted"
      # instead of network_mode below, could declare RCON_HOST
      # RCON_HOST: mc
    volumes:
    # mount the same volume used by server, but read-only
    - /opt/minecraft/cobalt:/data:ro
    # use a host attached directory so that it in turn can be backed up
    # to external/cloud storage
    - /mnt/betelgeuse/Backups/Minecraft:/backups
    # share network namespace with server to simplify rcon access
    restart: unless-stopped
    network_mode: "host
```

### Container logs

```
Paste logs here
```

### PR DESCRIPTION
**Pull request by itzg**
Fixes #1274

### CODE REVIEW

File: scripts/start-setupMounts [hunk 1]
```diff
@@ -24,8 +25,8 @@ else
 fi
 
 if [ -d /plugins ]; then
-  case ${TYPE} in
-    SPIGOT|BUKKIT|PAPER|MAGMA)
+  case ${FAMILY} in
+    SPIGOT|HYBRID)
```
Comment (shotah) [inline scripts/start-setupMounts:29]: excuse my ignorance, but where is HYBRID coming from?

Comment (itzg) [reply scripts/start-setupMounts:29]: It's a convention I had to make up for grouping all the various server types. Here's one example

https://github.com/itzg/docker-minecraft-server/blob/f29f098b2628666c50a2cb3448eecb61614fef6d/scripts/start-deployMagma#L92

Comment (shotah) [reply scripts/start-setupMounts:29]: I see it now. This looks good to me if you've tested it. 👍

Comment (itzg) [reply scripts/start-setupMounts:29]: Thanks! I'll go ahead and merge it since I had already forgotten about this PR 😊

**Pull request merged**